source/common/network/io_socket_handle_impl.cc

```diff
@@ -99,15 +99,16 @@ Api::IoCallUint64Result IoSocketHandleImpl::sendmsg(const Buffer::RawSlice* slic
     return sysCallResultToIoCallResult(result);
   } else {
     const size_t space_v6 = CMSG_SPACE(sizeof(in6_pktinfo));
-    // FreeBSD only needs in_addr size, but allocates more to unify code in two platforms.
     const size_t space_v4 = CMSG_SPACE(sizeof(in_pktinfo));
-    const size_t cmsg_space = (space_v4 < space_v6) ? space_v6 : space_v4;
+
+    // FreeBSD only needs in_addr size, but allocates more to unify code in two platforms.
+    const size_t cmsg_space = std::max(space_v6, space_v4);
     // kSpaceForIp should be big enough to hold both IPv4 and IPv6 packet info.
```

Comment:
What is "kSpaceForIp"?

IMO we should be using the amount we need based on v4/v6. This would work on Windows (and I would imagine the other platforms too)
`const size_t cmsg_space = (self_ip->version() == Address::IpVersion::v4) ? CMSG_SPACE(sizeof(in_pktinfo)) : CMSG_SPACE(sizeof(in6_pktinfo));`

But if there is some legitimate reason we need the larger of the two, just use `CMSG_SPACE(sizeof(in6_pktinfo))` and static_assert that `CMSG_SPACE(sizeof(in6_pktinfo)) >= CMSG_SPACE(sizeof(in_pktinfo))`

Reply:
Why not simply delay allocating cbuf to use message.msg_controllen as illustrated below and then assign msg_control?

Reply:
No strong opinion here, I totally agree with you my main motivation was not to make any functional change there unless there was a reason but I went for it now 0b477d9efcbf75b5a193b70ef7e441a33b8afb85.

(not relevant now) std:max is `constrexpr` since C++14 so the max function should be evaluated at compile time, this is why I preferred it over using `CMSG_SPACE(sizeof(in6_pktinfo))` and adding a `static_assert`.